[DOCS] Clarify remote_cluster_client is required to run ML (#72569)
elastic/elasticsearch: docs/reference/modules/node.asciidoc

node.roles: [ ml, remote_cluster_client] <1>
xpack.ml.enabled: true <2>
----

> <1> The `remote_cluster_client` role is optional but strongly recommended on all {ml} nodes.

Reviewer: Since this section is about ML nodes, I don't think this clause is necessary.
```suggestion
<1> The `remote_cluster_client` role is optional but strongly recommended.
```

node.roles: [ ml, remote_cluster_client] <1>
xpack.ml.enabled: true <2>
----
<1> The `remote_cluster_client` role is optional but strongly recommended on all {ml} nodes.
Otherwise, {ccs} fails when used in {ml} jobs or {dfeeds}. See <<remote-node>>.
<2> The `xpack.ml.enabled` setting is enabled by default.

> WARNING: The `remote_cluster_client` role is required on all master-eligible nodes in case

Reviewer: ```suggestion
WARNING: If you use {ccs} in your {anomaly-jobs}, the `remote_cluster_client` role is required on all master-eligible nodes.
```

Reviewer 2: correct, only if using ccs. We will only make the check if a ccs index is included in the datafeed.

> you are using {ccs}.

Reviewer: ```suggestion

```

> Otherwise, {ccs} fails when used in {ml} jobs or {dfeeds} against remote indices.

Reviewer: ```suggestion
Otherwise, the job or {dfeed} fails when it tries to access remote indices.
```

Author: The ML job will not start at all. The master doesn't access the remote indices, but it needs to validate license & permissions.

Reviewer 2: More like

> If you use CCS in anomaly detection jobs, the `remote_cluster_client` role is required on all master-eligible nodes or the datafeed will not start. It is also required on all data nodes, or CCS jobs will fail to query the required data.

In short, it should just be enabled on all data and master nodes :)

Reviewer 2: WOOPS! Forget that I ever mentioned data nodes. I meant `ML` nodes.

Reviewer: Thanks, that makes more sense. I've edited my suggestion.
